src/shadowfax.py

```diff
@@ -60,12 +60,30 @@ async def shadowfax(message, guid, logger):
     for node in qgraph["nodes"].values():
         if node.get("ids", None) is not None:
             pinned_node_ids.append(node["ids"][0])
-        else:
-            unpinned_node_category = node.get("categories", ["biolink:NamedThing"])[0]
     if len(pinned_node_ids) != 2:
         logger.info(f"{guid}: Pathfinder queries require two pinned nodes.")
         return message, 500
     
+    intermediate_categories = []
+    if qgraph.get("paths", None) is not None:
+        path_key = next(iter(qgraph["paths"].keys()))
+        qpath = qgraph["paths"][path_key]
+        if qpath.get("constraints", None):
+            constraints = qpath["constraints"]
+            if len(constraints) > 1:
+                logger.info(f"{guid}: Pathfinder queries do not support multiple constraints.")
+                return message, 500
+            if len(intermediate_categories) > 1:
+                logger.info(f"{guid}: Pathfinder queries do not support multiple intermediate categories")
+                return message, 500
+            if len(constraints) > 0:
+                intermediate_categories = constraints[0].get("intermediate_categories", None) or []
+        else:
+            intermediate_categories = ["biolink:NamedThing"]
+    else:
+        logger.info(f"{guid}: Pathfinder queries require paths.")
```

Comment:
This should probably be an error log instead of info. But also, haven't we already checked that this is a valid pathfinder in order to get into Shadowfax and this check is redundant?